pkg/utils/iptables.go

```diff
@@ -5,10 +5,42 @@ import (
 	"fmt"
 	"os/exec"
 	"strings"
+
+	"github.com/coreos/go-iptables/iptables"
+	v1core "k8s.io/api/core/v1"
 )
 
 var hasWait bool
 
+// Interface based on the IPTables struct from github.com/coreos/go-iptables
+// which allows to mock it.
+type IPTablesHandler interface {
```

Comment:
Is this abstraction necessary?

Reply:
Yes, it's needed for unit tests. Here is the fake implementation of that interface:

https://github.com/k3s-io/kube-router/pull/7/commits/1662197e41ba5ffe843f63f5600ee9c5c538db9f#diff-99cf22c0810de0362a6cb5c52ba159bc6297607a3bf86dab89d34d93e83f4507R597-R740

We don't want to call the real iptables in unit tests (apart from littering the system with rules, we would need to run unit tests as root).